FantasyUniverse/Services/QuestionDataConverter.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  QuestionViewData.swift
+//  FantasyUniverse
+//
+//  Created by TarasPeregrinus on 23.03.2022.
+//
+
+import Foundation
+import FirebaseFirestoreSwift
+
+enum QuestionDataConverter {
+    
+    static func formatQuestions(_ questions: [Question]) -> [QuestionViewData]? {
+        var questionViewData: [QuestionViewData]?
+        for question in questions {
+            // swiftlint:disable line_length
+            guard let questionString = question.question, let optionA = question.optionA, let optionB = question.optionB, let optionC = question.optionC, let optionD = question.optionD, let correct = question.correct
+            else {
+                if let missingValuesMessages = missingValuesMessages(of: question) {
+                    debugPrint(missingValuesMessages)
+                }
+                questionViewData = nil
```

Comment:
One empty question shouldn't be a reason to remove all the question data?

I'd just `continue` in this case.

Reply:
@dzoanb In this case we set the result to nil if the input array contains at least one element that does not match the expected question object structure, e.g. if one of input objects does not contain an essential element (a question itself or a correct answer value). So I thing that in such cases it is better not to create a QuestionCollection at all as there is something wrong with input data. Although I hope to discuss it with You.